portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,13 +46,21 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
     String comment = req.getParameter("comment");
     if (comment != null && !comment.equals("")) {
       // do nothing if the comment field is empty
-      comments.add(comment);
+      Entity commentEntity = new Entity("comment");
+      commentEntity.setProperty("value", comment);
+      datastore.put(commentEntity);
     }
     resp.sendRedirect("/index.html");
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("comment");
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      comments.add((String)entity.getProperty("value"));
+    }
     response.setContentType("application/json;");
     response.getWriter().println((new Gson()).toJson(comments));
```

Comment:
does toJson() accept an Iterable as an overload? If so, conversion to a list is unnecessary. If not, does PreparedQuery offer a list API?

Reply:
`toJson` does accept an iterable, however I thought it would be a good idea to extract the comment values using `entity.getProperty`.